src/Domain/ElementLogicalCoordinates.cpp

```diff
@@ -60,12 +60,12 @@ element_logical_coordinates(
     return std::nullopt;
   }
   const auto& block_search_tree = block_search_tree_it->second;
-  const auto found_element_id = block_search_tree.qbegin(
-      boost::geometry::index::covers(block_logical_coords.data));
-  if (found_element_id == block_search_tree.qend()) {
+  const auto found_element_ids =
```

Comment:
We only need the first element here, not all. Maybe just return the iterator like before?

Reply:
The iterator that is defined in the header that the entire point of this PR is to avoid including?